### ISSUE TITLE
Add a way to use predefined merry errors to wrap 

### ISSUE THREAD

**Issue by cosmicz** [ansel1/merry#21]
We generally define constants for often-used errors:
```go
var (
  ErrInvalidParameter = merry.
    New("Invalid parameter").
    WithUserMessage("Invalid parameter").
    WithHTTPCode(http.StatusBadRequest)
)
```
Then in the controller I'd love to be able to write something like:
```go
result, err := strconv.ParseBool(param)
if err != nil {
  // this should save the current stack and the err object, but maybe override the http code
  // and user/error message if present.
  return ErrInvalidParameter.Wrap(err)
}
```

**Comment by luveenw** [comment]
👍

**Comment by ansel1** [comment]
Have you checked out the Here() function?  It was intended to support that use case. Let me know if you had something else in mind.

**Comment by ansel1** [comment]
Wait, I get it. Pick up the props of the var  error, but also wrapping the error at the site. The issue is, merry errors can only have a single root cause error. Is the error you are returning an ErrInvalidParameter, or is it the strconv.ParseBool error?  Because it can’t really be both. 

In the sample you give, it sounds like the ErrInvalidParameter is just used as a template, to copy a set of common merry attributes onto the strconv error. But merry.Is(err, ErrInvalidParameter) would return false. 

If the desire is to have reusable error templates, you could achieve this with a function, like ‘func newErrInvalidParamete(err error) error { return merry.Wrap(err).WithMessage(...)... etc}’

If the desire is to link a causal chain of errors together, so that merry.Is() could return true if the error chain contains any of the causal errors, then that would need to be a new feature. Some other error packages have the notion of a “cause” property, and I’ve thought about adding that to merry, but I’ve never quite figured out an elegant API for it.

**Comment by ansel1** [comment]
Take a look at #13.  It's a proposal for a Cause feature.  See if it would suit your needs.  I'd probably also implement the `causer` interface from https://github.com/pkg/errors, since that is a widely used package.

**Comment by ansel1** [comment]
Fixed in 1.2.0
